includes/gateways/paypal/class-account-status-validator.php

```diff
@@ -0,0 +1,181 @@
+
+	/**
+	 * @var array Enabled webhook event names.
+	 */
+	public $enabled_webhook_events = array();
+
+	/**
+	 * @var \WP_Error Errors thrown when checking webhook status.
+	 */
+	public $errors_for_webhook;
+
+	/**
+	 * @var string Identifier of the connected account -- if available.
+	 */
+	public $connected_account = '';
+
+	/**
+	 * @var string Current store mode.
+	 */
+	private $mode;
+
+	/**
+	 * AccountStatusValidator constructor.
+	 *
+	 * @param string $mode Mode to check (`live` or `sandbox`). If omitted, current store mode is used.
+	 */
+	public function __construct( $mode = '' ) {
+		if ( empty( $mode ) ) {
+			$mode = edd_is_test_mode() ? PayPal\API::MODE_SANDBOX : PayPal\API::MODE_LIVE;
+		}
+		$this->mode = $mode;
+
+		// Set up base error objects.
+		$this->errors_for_credentials      = new \WP_Error();
+		$this->errors_for_merchant_account = new \WP_Error();
+		$this->errors_for_webhook          = new \WP_Error();
+	}
+
+	/**
+	 * Checks everything.
+	 *
+	 * @since 2.11
+	 */
+	public function check() {
+		$this->check_rest();
+		$this->check_merchant_account();
+		$this->check_webhook();
+	}
+
+	/**
+	 * Checks for valid REST API credentials.
+	 *
+	 * @since 2.11
+	 */
+	public function check_rest() {
+		try {
+			$api      = new API( $this->mode );
+			$response = $api->make_request( 'v1/identity/oauth2/userinfo?schema=paypalv1.1', array(), array(), 'GET' );
+
+			if ( empty( $response->user_id ) ) {
+				throw new \Exception( __( 'Unable to retrieve account information from PayPal. If the issue persists, try reconnecting.', 'easy-digital-downloads' ) );
+			}
+
+			$this->has_rest_credentials = true;
+		} catch ( Exceptions\Authentication_Exception $e ) {
+			$this->errors_for_credentials->add( 'no_credentials', __( 'Not connected.', 'easy-digital-downloads' ) );
+		} catch ( \Exception $e ) {
+			$this->errors_for_credentials->add( 'credentials_error', $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Determines if the merchant account is ready to accept payments.
+	 * It's possible (I think) to have valid API credentials ( @see AccountStatusValidator::check_rest() )
+	 * but still be unable to start taking payments, such as because your account
+	 * email hasn't been confirmed yet.
+	 *
+	 * @since 2.11
+	 */
+	public function check_merchant_account() {
+		try {
+			$this->merchant_details = MerchantAccount::retrieve();
+			$this->merchant_details->validate();
+
+			if ( ! $this->merchant_details->is_account_ready() ) {
+				foreach ( $this->merchant_details->get_errors()->get_error_codes() as $code ) {
+					$this->errors_for_merchant_account->add( $code, $this->merchant_details->get_errors()->get_error_message( $code ) );
+				}
+			}
+		} catch ( Exceptions\MissingMerchantDetails $e ) {
+			$this->errors_for_merchant_account->add( 'missing_merchant_details', __( 'Missing merchant details from PayPal. Please reconnect and make sure you click the button to be redirected back to your site.', 'easy-digital-downloads' ) );
```

Comment:
I'm getting this when I'm partially connected...obviously I need to disconnect and then reconnect but should the message include the word disconnect?  Because I might be tempted to attempt to short-circuit the connection issue and assume that I could click the "Sync Webhook" button and fix everything 😇 and then I get a funny error. Maybe the "Sync Webhook" button should be disabled, if possible?

Reply:
Sync Webhook actually only requires API credentials; it doesn't require these merchant details. So that button will still work as long as the API connection is valid. (Your error is just because there are no changes to make and I'll fix that separately.)